normalizer/core.py

```diff
@@ -370,6 +370,7 @@ def normalize(
     env: Dict = {},
     build_env: Dict = {},
     dry_run: bool = False,
+    dockerfile: str = None,
```

Comment:
```suggestion
    dockerfile: str = '',
```

Reply:
If we change `dockerfile: str = None`  ro `dockerfile: str = ''`.
We determine whether there is only by `dockerfile == ''`, It's so inelegant.

Reply:
Isn't that `''` also false?
You specified dockerfile to be `str` then it should not be None.
Or else use `Optional[str]`

Reply:
> Isn't that '' also false?

You are right: '' is also false.